CRM/Utils/Token.php

```diff
@@ -1404,7 +1404,7 @@ public static function replaceGreetingTokens(&$tokenString, $contactDetails = NU
         $greetingTokens = $remainingTokens;
         reset($greetingTokens);
         $greetingsReturnProperties = array();
-        while (list($key) = each($greetingTokens)) {
+        foreach ($greetingTokens as $key => $value) {
           $props = array_flip(CRM_Utils_Array::value($key, $greetingTokens));
```

Comment:
Or 
```patch
- foreach ($greetingTokens as $key => $value) {
-    $props = array_flip(CRM_Utils_Array::value($key, $greetingTokens));
+ foreach ($greetingTokens as $value) {
+   $props = array_flip($value);
```
??
As $key is only used in 1 place and $value is no longer used

Reply:
hmm that could work i guess i was just aiming at a minimalist replacement process but that could work